tests/RobotFramework/tests/cumulocity/configuration/legacy_config_operation/legacy_configuration_workflow.robot

```diff
@@ -0,0 +1,54 @@
+*** Settings ***
+Resource            ../../../../resources/common.resource
+Library             String
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Test Setup          Custom Setup
+Test Teardown       Get Logs
+
+Test Tags           theme:c8y    theme:configuration
+
+
+*** Test Cases ***
+Support Legacy Configuration Download Operation
+    [Documentation]    Support Cumulocity Legacy File Upload/Download
+    ...    which was the default before file-type configuration operations were introduced
+    ...    Cumulocity Docs: https://cumulocity.com/docs/device-integration/fragment-library/#install-legacy-configuration
+
+    # pre-condition
+    Cumulocity.Managed Object Should Not Have Fragments    c8y_ConfigurationDump
+
+    # Send configuration to a device
+    ${binary_url}=    Create Inventory Binary    dummy    example    contents=foobar
+    ${binary_id}=    Get Binary ID From URL    ${binary_url}
+    ${operation}=    Cumulocity.Create Operation
+    ...    description=Legacy Configuration Download
+    ...    fragments={"c8y_DownloadConfigFile":{"url": "${binary_url}", "c8y_ConfigurationDump": {"id": "${binary_id}"}}}
+    Cumulocity.Operation Should Be SUCCESSFUL    ${operation}
+
+    # Server will update the inventory managed object fields
+    Cumulocity.Device Should Have Fragments    c8y_ConfigurationDump
+    Cumulocity.Device Should Have Fragment Values    c8y_ConfigurationDump.id\="${binary_id}"
+
+    # Get configuration from the device
+    ${operation}=    Cumulocity.Create Operation
+    ...    description=Legacy Configuration Upload
+    ...    fragments={"c8y_UploadConfigFile":{}}
+    Cumulocity.Operation Should Be SUCCESSFUL    ${operation}
```

Comment:
As per the documentation, creating a managed object of type `c8y_ConfigurationDump` is required as part of the `c8y_UploadConfigFile` operation flow as it is **not done implicitly**, as done for the `c8y_DownloadConfigFile` operation (at least as per the doc). Here are the relevant excerpts from the doc describing the same(Note step 4):

>When uploading its configuration, the device must first upload the configuration file into Cumulocity inventory binaries, and then create a configuration repository entry as a managed object of type c8y_ConfigurationDump in the inventory. This object must then contain a link to the just uploaded file.

<blockquote>
SmartREST example

1. Receive c8y_UploadConfigFile operation: 520,DeviceSerial
2. Set operation status to EXECUTING: 501,c8y_UploadConfigFile
3. Upload configuration to inventory binaries API using REST
4. Create configuration repository entry in inventory using REST
5. Set operation status to SUCCESSFUL: 503,c8y_UploadConfigFile
</blockquote>

Reply:
The `c8y_ConfigurationDump` is not required, or at least not in the system test as this is added by the operation. We're not really interesting in reproducing the entire server behaviour here as that contract is required by the UI component only.

Reply:
if so, then I'll update the PR description with the note about the file upload and saying that thin-edge.io does not adhere to the legacy contract as thin-edge.io already supports requesting the "default" type via the type-based operation.

Reply:
The user provided feedback and is ok with thin-edge.io not adhering to the full legacy config upload handling.